### DISCUSSION TITLE
Solving commit issues when syncing between Linux and MacOS

### DISCUSSION THREAD

**Discussion by FlipH19Switch** [CharlesChiuGit/Logseq-Git-Sync-101#36] [General]
Thought I'd add this here in case anyone else has the same problem.

I ran into an issue with commits not getting pushed from MacOS. I initially set everything up on a machine running Linux Mint, confirmed it was auto-uploading to Github, then cloned the repo to a M1 laptop running MacOS Sonoma and began testing there.

My auto-commits on MacOS were not being pushed to GitHub. The problem was that I assumed the pre-commit and post-commit files I added on Linux would be synced automatically and appear on MacOS. Once I added these files to the .git/hooks folder on MacOS and ran the necessary command to make them executable, files began uploading correctly.

Now everything works on both machines :)

**Comment by charliie-dev** [comment]
yes, files in `.git` folder won't be synced to the remote git repo, that's why I put them in the `git-hooks` folder of this repo.
glad to hear you solved it! nice!
